daemon/config/config_unix.go

```diff
@@ -61,3 +68,24 @@ func (conf *Config) IsSwarmCompatible() error {
 	}
 	return nil
 }
+
+func verifyDefaultIpcMode(mode string) error {
+	const hint = "Use \"shareable\" or \"private\"."
+
+	dm := containertypes.IpcMode(mode)
+	if !dm.Valid() {
+		return fmt.Errorf("Default IPC mode setting (%v) is invalid. "+hint, dm)
+	}
+	if dm != "" && !dm.IsPrivate() && !dm.IsShareable() {
+		return fmt.Errorf("IPC mode \"%v\" is not supported as default value. "+hint, dm)
```

Comment:
I'd rather leave it as is, since the error diagnostics is cleaner if we distinguish between an invalid value and a value that is valid but can not be used as default. Surely, it's a minor thing so whatever you say.